game/world/objects/npc.cpp

```diff
@@ -1352,6 +1367,16 @@ bool Npc::implTurnTo(float dx, float dz, bool noAnim, uint64_t dt) {
   return rotateTo(dx,dz,step,noAnim,dt);
   }
 
+bool Npc::implWhirlTo(const Npc &oth, uint64_t dt) {
+  if(&oth==this)
+    return true;
+  auto dx = oth.x-x;
+  auto dz = oth.z-z;
+  auto  gl   = guild();
+  float step = float(owner.script().guildVal().turn_speed[gl]) * 2;
```

Comment:
Based on video evidence you shared previously: 
> https://youtu.be/GKpt0Y07gJw?t=1538 ... when talking to Xardas' daemon, which is this sequence from above

Rotation speed appears(not 100% sure) to be the same, but with different animation.

Reply:
Ahhhh ... you're right ... thanks for noticing that discrepancy.

Digging deeper, the animation data does have t_surprise_ccw and t_surprise_cw states and G1 data comes uncompiled with "//WhirlAround" comments intact above those :-) .

Additionally, it looks like in G1 everybody including humans have the whirl animation, where in G2, monsters have it - including the newly added ones - but not humans.

So I'll continue digging :-) .

Reply:
People are whirling really nicely now and the code looks quite a bit different :-)